operations/mimir/autoscaling.libsonnet

```diff
@@ -139,6 +147,40 @@
     ],
   }),
 
+  local newQueryFrontendScaledObject(name, cpu_requests, memory_requests, min_replicas, max_replicas) = self.newScaledObject(
```

Comment:
This is getting quite duplicated among all components scaled by CPU and memory. I **prefer keeping it as is in this PR** but I would be glad if you could attempt a refactoring in a follow up to have a generic `newCPUAndMemoryScaledObject()` that we can just reused among components scaled by CPU and memory.

Reply:
Yep, I acknowledge we should refactor this code :)